controllers/kubeadmconfig_controller.go

```diff
@@ -79,6 +78,11 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 
+	// ignore machines not in pending state
```

Comment:
We said in the proposal that providers should not look at machine phase to determine what to do. What you can do here, instead, is ignore machines that have a non-nil spec.bootstrap.data value.

Reply:
Or you could look at machine.status.bootstrapReady.

Reply:
ah...ok...yes, this is where we look at the real data instead of the field that is generated by the real data. Right. I remember that discussion now.

Reply:
I suspect that many providers will simply read the phase since it is available...

maybe the machine should do the reading and give me the phase, like

```
machine.Phase() == Pending
```

which looks directly at the data.

Thoughts @ncdc?